src/app/pages/data-protection/snapshot/snapshot-form/snapshot-form.component.ts

```diff
@@ -1,244 +1,151 @@
-import { Component } from '@angular/core';
-import { AbstractControl, Validators } from '@angular/forms';
+import {
+  ChangeDetectionStrategy, ChangeDetectorRef, Component,
+} from '@angular/core';
+import { Validators } from '@angular/forms';
+import { FormBuilder } from '@ngneat/reactive-forms';
 import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
-import * as _ from 'lodash';
+import { TranslateService } from '@ngx-translate/core';
+import { Observable, of } from 'rxjs';
+import { LifetimeUnit } from 'app/enums/lifetime-unit.enum';
 import helptext from 'app/helptext/data-protection/snapshot/snapshot-form';
-import { FormConfiguration } from 'app/interfaces/entity-form.interface';
 import { PeriodicSnapshotTask } from 'app/interfaces/periodic-snapshot-task.interface';
-import { FieldSets } from 'app/modules/entity/entity-form/classes/field-sets';
-import { EntityFormComponent } from 'app/modules/entity/entity-form/entity-form.component';
-import { FormSelectConfig, UnitType } from 'app/modules/entity/entity-form/models/field-config.interface';
-import { EntityUtils } from 'app/modules/entity/utils';
+import { FormErrorHandlerService } from 'app/modules/ix-forms/services/form-error-handler.service';
+import { crontabToSchedule } from 'app/modules/scheduler/utils/crontab-to-schedule.utils';
 import { CronPresetValue } from 'app/modules/scheduler/utils/get-default-crontab-presets.utils';
-import { DialogService, StorageService, TaskService } from 'app/services';
-import { ModalService } from 'app/services/modal.service';
+import { scheduleToCrontab } from 'app/modules/scheduler/utils/schedule-to-crontab.utils';
+import { StorageService, TaskService, WebSocketService } from 'app/services';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
 
 @UntilDestroy()
 @Component({
-  selector: 'app-cron-snapshot-task-add',
-  template: '<entity-form [conf]="this"></entity-form>',
-  providers: [TaskService],
+  templateUrl: './snapshot-form.component.html',
+  styleUrls: ['./snapshot-form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SnapshotFormComponent implements FormConfiguration {
-  queryKey = 'id';
-  queryCall = 'pool.snapshottask.query' as const;
-  addCall = 'pool.snapshottask.create' as const;
-  editCall = 'pool.snapshottask.update' as const;
-  isEntity = true;
-  pk: number;
-  protected dataset: string;
-  protected isDatasetDisabled = false;
-  protected datasetFg: AbstractControl;
-  saveButtonEnabled = true;
-  protected entityForm: EntityFormComponent;
-  title: string;
-  isNew = false;
+export class SnapshotFormComponent {
```

Comment:
Let's also rename SnapshotFormComponent and folders to `snapshot-task` to make it less confusing.